graph/cytoscape/cytoscape.go

```diff
@@ -0,0 +1,200 @@
+// cytoscape provides conversion from graph/tree to cystoscape configuration json
+package cytoscape
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/swift-sunshine/swscore/graph/tree"
+	"github.com/swift-sunshine/swscore/log"
+)
+
+type NodeData struct {
+	Id            string `json:"id"`
+	Service       string `json:"service"`
+	Version       string `json:"version,omitempty"`
+	Text          string `json:"text"`
+	Parent        string `json:"parent,omitempty"`
+	LinkPromGraph string `json:"link_prom_graph,omitempty"`
+}
+
+type EdgeData struct {
+	Id     string `json:"id"`
+	Source string `json:"source"`
+	Target string `json:"target"`
+	Text   string `json:"text"`
+	Color  string `json:"color"`
+	Rpm    string `json:"req_per_min,omitempty"`
+	Rpm2xx string `json:"req_per_min_2XX,omitempty"`
+	Rpm3xx string `json:"req_per_min_3XX,omitempty"`
+	Rpm4xx string `json:"req_per_min_4XX,omitempty"`
+	Rpm5xx string `json:"req_per_min_5XX,omitempty"`
+}
+
+type NodeWrapper struct {
+	Data NodeData `json:"data"`
+}
+
+type EdgeWrapper struct {
+	Data EdgeData `json:"data"`
+}
+
+type Elements struct {
+	Nodes []*NodeWrapper `json:"nodes"`
+	Edges []*EdgeWrapper `json:"edges"`
+}
+
+type Config struct {
```

Comment:
I guess that this model is a mapping between cytoscape model (or is it defined by us?).
If it responds of how cytoscape expects a json model it would be nice to have some reference to the schema used, just to read it if we need to make any change and understand the mappings.

Reply:
see updated documentation in package files, I think it will answer your questions.